Subject: Disable <section> behaves as <hr>
Author: Tatsuya Kinoshita <tats@debian.org>

diff --git a/file.c b/file.c
index 78987e9..1ab1c55 100644
--- a/file.c
+++ b/file.c
@@ -4838,7 +4838,6 @@ HTMLtagproc1(struct parsed_tag *tag, struct html_feed_environ *h_env)
 	}
 	flushline(h_env, obuf, envs[h_env->envc].indent, 0, h_env->limit);
 	return 0;
-    case HTML_SECTION:
     case HTML_HR:
 	close_anchor(h_env, obuf);
 	tmp = process_hr(tag, h_env->limit, envs[h_env->envc].indent);
