Description: fix spelling error
Author: HIGUCHI Daisuke (VDR dai) <dai@debian.org>
Last-Update: 2017-12-24

Index: w3m/scripts/w3mman/w3mman.1.in
===================================================================
--- w3m.orig/scripts/w3mman/w3mman.1.in
+++ w3m/scripts/w3mman/w3mman.1.in
@@ -67,7 +67,7 @@ If W3MMAN_MAN is set, its value is used
 .SH FILES
 .TP
 .I @libexecdir@/@PACKAGE@/cgi-bin/w3mman2html.cgi
-manual page to HTML convertor.
+manual page to HTML converter.
 .SH SEE ALSO
 .BR man (1),
 .BR w3m (1).
Index: w3m/terms.c
===================================================================
--- w3m.orig/terms.c
+++ w3m/terms.c
@@ -823,7 +823,7 @@ ttymode_set(int mode, int imode)
     while (TerminalSet(tty, &ioval) == -1) {
 	if (errno == EINTR || errno == EAGAIN)
 	    continue;
-	printf("Error occured while set %x: errno=%d\n", mode, errno);
+	printf("Error occurred while set %x: errno=%d\n", mode, errno);
 	reset_error_exit(SIGNAL_ARGLIST);
     }
 #endif
@@ -844,7 +844,7 @@ ttymode_reset(int mode, int imode)
     while (TerminalSet(tty, &ioval) == -1) {
 	if (errno == EINTR || errno == EAGAIN)
 	    continue;
-	printf("Error occured while reset %x: errno=%d\n", mode, errno);
+	printf("Error occurred while reset %x: errno=%d\n", mode, errno);
 	reset_error_exit(SIGNAL_ARGLIST);
     }
 #endif /* __MINGW32_VERSION */
@@ -861,7 +861,7 @@ set_cc(int spec, int val)
     while (TerminalSet(tty, &ioval) == -1) {
 	if (errno == EINTR || errno == EAGAIN)
 	    continue;
-	printf("Error occured: errno=%d\n", errno);
+	printf("Error occurred: errno=%d\n", errno);
 	reset_error_exit(SIGNAL_ARGLIST);
     }
 }
@@ -2282,7 +2282,7 @@ sleep_till_anykey(int sec, int purge)
     }
     er = TerminalSet(tty, &ioval);
     if (er == -1) {
-	printf("Error occured: errno=%d\n", errno);
+	printf("Error occurred: errno=%d\n", errno);
 	reset_error_exit(SIGNAL_ARGLIST);
     }
     return ret;
