Description: Appease gcc -Werror=format-security
Author: Colin Watson <cjwatson@ubuntu.com>
Forwarded: no
Last-Update: 2011-10-23

Index: b/acinclude.m4
===================================================================
--- a/acinclude.m4
+++ b/acinclude.m4
@@ -866,7 +866,7 @@
 AC_TRY_COMPILE(
 changequote(<<,>>)dnl
 <<extern char *sys_errlist[];>>,
-<<printf(sys_errlist[0]);>>,
+<<printf("%s", sys_errlist[0]);>>,
 changequote([,])dnl
 [have_sys_errlist="yes"; AC_DEFINE(HAVE_SYS_ERRLIST)],
 [have_sys_errlist="no"])
Index: b/configure
===================================================================
--- a/configure
+++ b/configure
@@ -8986,7 +8986,7 @@
 int
 main ()
 {
-printf(sys_errlist[0]);
+printf("%s", sys_errlist[0]);
   ;
   return 0;
 }
Index: b/w3mbookmark.c
===================================================================
--- a/w3mbookmark.c
+++ b/w3mbookmark.c
@@ -99,7 +99,7 @@
 	fprintf(f, "<body>\n<h1>Bookmarks</h1>\n");
 	fprintf(f, "<h2>%s</h2>\n<ul>\n", section);
 	fprintf(f, "<li><a href=\"%s\">%s</a>\n", url, title);
-	fprintf(f, end_section);
+	fprintf(f, "%s", end_section);
 	fprintf(f, "</ul>\n</body>\n</html>\n");
 	fclose(f);
     }
