Description: Support Boehm GC 7.2
Origin: https://bugs.debian.org/cgi-bin/bugreport.cgi?msg=5;filename=080_gc72.patch;att=1;bug=758831
From: Micah Cowan <micah@addictivecode.org>
Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=758831
Bug-Fedora: https://bugzilla.redhat.com/show_bug.cgi?id=555467
Bug: http://sourceforge.net/p/w3m/patches/63/
Bug: http://sourceforge.net/p/w3m/patches/59/

Index: w3m-0.5.3/main.c
===================================================================
--- w3m-0.5.3.orig/main.c	2014-08-21 12:22:32.000000000 -0700
+++ w3m-0.5.3/main.c	2014-08-21 12:22:52.419420062 -0700
@@ -845,7 +845,12 @@ main(int argc, char **argv, char **envp)
     mySignal(SIGPIPE, SigPipe);
 #endif
 
+#if GC_VERSION_MAJOR >= 7 && GC_VERSION_MINOR >= 2
+    orig_GC_warn_proc = GC_get_warn_proc();
+    GC_set_warn_proc(wrap_GC_warn_proc);
+#else
     orig_GC_warn_proc = GC_set_warn_proc(wrap_GC_warn_proc);
+#endif
     err_msg = Strnew();
     if (load_argc == 0) {
 	/* no URL specified */
