Subject: Update contact list in README
From: Tatsuya Kinoshita <tats@debian.org>
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=696209

diff --git a/doc-jp/README b/doc-jp/README
index 074766f..e5e59b1 100644
--- a/doc-jp/README
+++ b/doc-jp/README
@@ -122,7 +122,6 @@ w3m 
 	東北大学病院
 	satodai@w3m.jp
 
-ご意見，ご感想をMLまでお寄せください．
-	w3m-dev@sic.med.tohoku.ac.jp (日本語)
-	w3m-dev-en@sic.med.tohoku.ac.jp (英語)
+ご意見，ご感想をお寄せください．
 	http://w3m.sourceforge.net/
+	https://sourceforge.net/projects/w3m/
diff --git a/doc/README b/doc/README
index 324ea3e..60d6d19 100644
--- a/doc/README
+++ b/doc/README
@@ -115,7 +115,6 @@ Current Maintainer
 	Tohoku University Hospital
 	satodai@w3m.jp
 
-Feel free to send your opinion to the w3m mailing-lists.
-	w3m-dev@sic.med.tohoku.ac.jp (Japanese)
-	w3m-dev-en@sic.med.tohoku.ac.jp (English)
+Feel free to send your opinion to:
 	http://w3m.sourceforge.net/
+	https://sourceforge.net/projects/w3m/
