Subject: Update contact list in README
From: Tatsuya Kinoshita <tats@debian.org>
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=696209

diff --git a/doc-jp/README b/doc-jp/README
index 074766f..6fe07a8 100644
--- a/doc-jp/README
+++ b/doc-jp/README
@@ -123,6 +123,6 @@ w3m 
 	satodai@w3m.jp
 
 ご意見，ご感想をMLまでお寄せください．
-	w3m-dev@sic.med.tohoku.ac.jp (日本語)
-	w3m-dev-en@sic.med.tohoku.ac.jp (英語)
+	w3m-dev@w3m.jp (日本語)
+	w3m-dev-en@w3m.jp (英語)
 	http://w3m.sourceforge.net/
diff --git a/doc/README b/doc/README
index 324ea3e..e468081 100644
--- a/doc/README
+++ b/doc/README
@@ -116,6 +116,6 @@ Current Maintainer
 	satodai@w3m.jp
 
 Feel free to send your opinion to the w3m mailing-lists.
-	w3m-dev@sic.med.tohoku.ac.jp (Japanese)
-	w3m-dev-en@sic.med.tohoku.ac.jp (English)
+	w3m-dev@w3m.jp (Japanese)
+	w3m-dev-en@w3m.jp (English)
 	http://w3m.sourceforge.net/
