Description: Support for TLS SNI (Server Name Indication)
Origin: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=523159#5
Author: Sascha Silbe <sascha-debian-bugs-w3m-1@silbe.org>
Bug-Debian: http://bugs.debian.org/523159

diff -ur w3m-0.5.2/url.c w3m-0.5.2-sni/url.c
--- w3m-0.5.2/url.c	2007-05-23 17:06:06.000000000 +0200
+++ w3m-0.5.2-sni/url.c	2009-03-11 17:44:51.210620186 +0100
@@ -374,6 +374,9 @@
 #if SSLEAY_VERSION_NUMBER >= 0x00905100
     init_PRNG();
 #endif				/* SSLEAY_VERSION_NUMBER >= 0x00905100 */
+#if (SSLEAY_VERSION_NUMBER >= 0x00908070) && !defined(OPENSSL_NO_TLSEXT)
+    SSL_set_tlsext_host_name(handle,hostname);
+#endif				/* (SSLEAY_VERSION_NUMBER >= 0x00908070) && !defined(OPENSSL_NO_TLSEXT) */
     if (SSL_connect(handle) > 0) {
 	Str serv_cert = ssl_get_certificate(handle, hostname);
 	if (serv_cert) {
