Origin: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=725892
Bug: https://sourceforge.net/p/w3m/patches/67/
From: Rafael Laboissiere <rafael@laboissiere.net>
Date: Wed, 9 Oct 2013 13:43:31 -0300
Subject: README.img: Fix typo

---
 doc/README.img | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/doc/README.img b/doc/README.img
index 0c10114..f11362f 100644
--- a/doc/README.img
+++ b/doc/README.img
@@ -41,7 +41,7 @@ Key functions
   Specify the following keymaps in ~/.w3m/keymap.
       keymap  X    DISPLAY_IMAGE
       keymap  C-c  STOP_IMAGE
-      keyamp  t    SET_OPTION  display_image=toggle
+      keymap  t    SET_OPTION  display_image=toggle
 
 Commandline options
 
-- 
1.8.4.rc3

