Subject: Fix crash after SEARCH_NEXT
From: AIDA Shinra <shinra@j10n.org>
Origin: http://www.j10n.org/files/w3m-cvs-1.1055-search-next.patch
    
    Patch from [w3m-dev:04473] on 2013-12-07.

--- w3m.orig/main.c	Mon Oct 14 02:20:37 2013
+++ w3m/main.c	Sat Dec  7 00:33:36 2013
@@ -1928,6 +1928,10 @@
     result = srchcore(SearchString, routine[reverse]);
     if (result & SR_FOUND)
 	clear_mark(Currentbuf->currentLine);
+    else {
+	if (reverse == 0)
+	    Currentbuf->pos -= 1;
+    }
     displayBuffer(Currentbuf, B_NORMAL);
     disp_srchresult(result, (reverse ? "Backward: " : "Forward: "),
 		    SearchString);
