Subject: Use pkg-config to build with imlib2 1.4.6
From: Tatsuya Kinoshita <tats@debian.org>

diff --git a/acinclude.m4 b/acinclude.m4
index e4ccc3d..56d6338 100644
--- a/acinclude.m4
+++ b/acinclude.m4
@@ -649,6 +649,9 @@ AC_DEFUN([AC_W3M_IMAGE],
      fi;;
    imlib2)
      with_imlib2="yes"
+     if test x"$PKG_CONFIG" = x; then
+       PKG_CONFIG=pkg-config
+     fi
      if test x"$IMLIB2_CONFIG" = x; then
        IMLIB2_CONFIG=imlib2-config
      fi;;
@@ -728,7 +731,7 @@ AC_DEFUN([AC_W3M_IMAGE],
      IMGTARGETS="x11"    
      AC_DEFINE(USE_IMLIB2)
      IMGX11CFLAGS="`${IMLIB2_CONFIG} --cflags`"
-     IMGX11LDFLAGS="`${IMLIB2_CONFIG} --libs`"
+     IMGX11LDFLAGS="-lX11 `${PKG_CONFIG} --libs imlib2`"
    else
      AC_MSG_WARN([unable to build w3mimgdisplay with X11 support])
    fi
@@ -756,7 +759,7 @@ AC_DEFUN([AC_W3M_IMAGE],
      AC_DEFINE(USE_IMLIB2)
      IMGOBJS="$IMGOBJS fb/fb_w3mimg.o fb/fb.o fb/fb_img.o"
      IMGFBCFLAGS="`${IMLIB2_CONFIG} --cflags`"
-     IMGFBLDFLAGS="`${IMLIB2_CONFIG} --libs`"
+     IMGFBLDFLAGS="`${PKG_CONFIG} --libs imlib2`"
    else
      AC_MSG_WARN([unable to build w3mimgdisplay with FB support])
    fi
diff --git a/configure b/configure
index 4c1bc06..01153c1 100755
--- a/configure
+++ b/configure
@@ -6811,6 +6811,9 @@ $as_echo "$with_imagelib" >&6; }
      fi;;
    imlib2)
      with_imlib2="yes"
+     if test x"$PKG_CONFIG" = x; then
+       PKG_CONFIG=pkg-config
+     fi
      if test x"$IMLIB2_CONFIG" = x; then
        IMLIB2_CONFIG=imlib2-config
      fi;;
@@ -6955,7 +6958,7 @@ $as_echo "$as_me: WARNING: Imlib2 is not installed.  Install Imlib2 (version >=
      $as_echo "#define USE_IMLIB2 1" >>confdefs.h
 
      IMGX11CFLAGS="`${IMLIB2_CONFIG} --cflags`"
-     IMGX11LDFLAGS="`${IMLIB2_CONFIG} --libs`"
+     IMGX11LDFLAGS="-lX11 `${PKG_CONFIG} --libs imlib2`"
    else
      { $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: unable to build w3mimgdisplay with X11 support" >&5
 $as_echo "$as_me: WARNING: unable to build w3mimgdisplay with X11 support" >&2;}
@@ -6991,7 +6994,7 @@ $as_echo "$as_me: WARNING: unable to build w3mimgdisplay with X11 support" >&2;}
 
      IMGOBJS="$IMGOBJS fb/fb_w3mimg.o fb/fb.o fb/fb_img.o"
      IMGFBCFLAGS="`${IMLIB2_CONFIG} --cflags`"
-     IMGFBLDFLAGS="`${IMLIB2_CONFIG} --libs`"
+     IMGFBLDFLAGS="`${PKG_CONFIG} --libs imlib2`"
    else
      { $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: unable to build w3mimgdisplay with FB support" >&5
 $as_echo "$as_me: WARNING: unable to build w3mimgdisplay with FB support" >&2;}
