Subject: Disable USE_EGD for LibreSSL
Author: Vsevolod Stakhov <vsevolod@FreeBSD.org>
Origin: https://bz-attachments.freebsd.org/attachment.cgi?id=144635
Bug-FreeBSD: https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=191852
Bug-FreeBSD: https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=191956

diff --git a/config.h.in b/config.h.in
index a4110ea..3b575e4 100644
--- a/config.h.in
+++ b/config.h.in
@@ -85,7 +85,7 @@
 #undef INET6
 #undef HAVE_SOCKLEN_T
 #undef HAVE_OLD_SS_FAMILY
-#define USE_EGD
+#undef USE_EGD
 #define ENABLE_REMOVE_TRAILINGSPACES
 #undef MENU_THIN_FRAME
 #undef USE_RAW_SCROLL
