Subject: Fix crash in parse_tag() during every start
Author: Olaf Hering <olh@suse.de>
Origin: https://build.opensuse.org/package/view_file/openSUSE:Factory/w3m/w3m-parsetagx-crash.patch?expand=1

diff --git a/parsetagx.c b/parsetagx.c
index 6b627d2..3435f9f 100644
--- a/parsetagx.c
+++ b/parsetagx.c
@@ -221,6 +221,7 @@ parse_tag(char **s, int internal)
          int j, hidden=FALSE;
          for (j=0; j<i; j++) {
            if (tag->attrid[j] == ATTR_TYPE &&
+               tag->value[j] &&
                strcmp("hidden",tag->value[j]) == 0) {
              hidden=TRUE;
              break;
