Subject: Correct printf arguments (still support <C99)
From: Scarlett <scarlett@xavin.net>, Tatsuya Kinoshita <tats@debian.org>
Origin: http://marc.info/?l=openbsd-ports&m=142090828929750&w=2

diff --git a/cookie.c b/cookie.c
index 8020f6d..c11649c 100644
--- a/cookie.c
+++ b/cookie.c
@@ -247,7 +247,7 @@ find_cookie(ParsedURL *pu)
 		Strcat(tmp, Sprintf("; $Domain=\"%s\"", p1->domain->ptr));
 	    if (p1->portl)
 		Strcat(tmp,
-		       Sprintf("; $Port=\"%s\"", portlist2str(p1->portl)));
+		       Sprintf("; $Port=\"%s\"", portlist2str(p1->portl)->ptr));
 	}
     }
     return tmp;
@@ -463,7 +463,7 @@ save_cookies(void)
 	    continue;
 	fprintf(fp, "%s\t%s\t%s\t%ld\t%s\t%s\t%d\t%d\t%s\t%s\t%s\n",
 		parsedURL2Str(&p->url)->ptr,
-		p->name->ptr, p->value->ptr, p->expires,
+		p->name->ptr, p->value->ptr, (long)p->expires,
 		p->domain->ptr, p->path->ptr, p->flag,
 		p->version, str2charp(p->comment),
 		(p->portl) ? portlist2str(p->portl)->ptr : "",
diff --git a/main.c b/main.c
index 32f4da2..76256d6 100644
--- a/main.c
+++ b/main.c
@@ -5841,7 +5841,7 @@ set_buffer_environ(Buffer *buf)
 	    set_environ("W3M_CURRENT_FORM", form2str((FormItemList *)a->url));
 	else
 	    set_environ("W3M_CURRENT_FORM", "");
-	set_environ("W3M_CURRENT_LINE", Sprintf("%d",
+	set_environ("W3M_CURRENT_LINE", Sprintf("%ld",
 						l->real_linenumber)->ptr);
 	set_environ("W3M_CURRENT_COLUMN", Sprintf("%d",
 						  buf->currentColumn +
diff --git a/map.c b/map.c
index 12701e7..bb240ea 100644
--- a/map.c
+++ b/map.c
@@ -573,7 +573,7 @@ page_info_panel(Buffer *buf)
 		   "<tr valign=top><td nowrap>Number of lines<td>",
 		   Sprintf("%d", all)->ptr,
 		   "<tr valign=top><td nowrap>Transferred bytes<td>",
-		   Sprintf("%d", buf->trbyte)->ptr, NULL);
+		   Sprintf("%lu", (unsigned long)buf->trbyte)->ptr, NULL);
 
     a = retrieveCurrentAnchor(buf);
     if (a != NULL) {
