Subject: Update ChangeLog
From: Tatsuya Kinoshita <tats@debian.org>

diff --git a/ChangeLog b/ChangeLog
index 88358ef..7629962 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,3 +1,605 @@
+2015-01-24  Tatsuya Kinoshita  <tats@debian.org>
+
+	* acinclude.m4, configure, version.c.in:
+	Update to 0.5.3+gitYYYYMMDD (generate from ChangeLog).
+
+2015-01-15  Tatsuya Kinoshita  <tats@debian.org>
+
+	* alloc.h, main.c: Drop C99 features.
+
+2015-01-15  Scarlett  <scarlett@xavin.net>
+
+	Add overflow detection.
+	Origin: http://marc.info/?l=openbsd-ports&m=142090828929750&w=2
+	* main.c: Call exit(1) when out of memory to avoid dereferencing null
+	pointers when gc's malloc fails.
+	* alloc.h: Replacements for w3m's allocation macros which add
+	overflow detection and concentrate the macros in one file.
+	* indep.h, libwc/charset.c, libwc/status.c, matrix.c: Use the
+	overflow-detecting allocation macros from alloc.h.
+
+2015-01-15  Tatsuya Kinoshita  <tats@debian.org>
+
+	* Str.c, cookie.c, map.c:
+	Do not use C99 printf format specifiers and asprintf.
+
+2015-01-15  Scarlett  <scarlett@xavin.net>
+
+	Correct printf arguments and use asprintf.
+	Origin: http://marc.info/?l=openbsd-ports&m=142090828929750&w=2
+	* Str.c: Use asprintf() instead of rolling our own printf string
+	length detection.
+	* cookie.c: Pass the char pointer in the string struct to printf %s
+	instead of the string struct itself.
+	Print time_t using %lld instead of %ld to allow for 64-bit time_t.
+	* main.c: Print a long int using the correct format specifier.
+	* map.c: Print size_t using the correct format specifier.
+
+2014-12-06  Araki Ken  <arakiken@users.sf.net>
+
+	Support OSC 5379 remote imaging and sixel graphics.
+	Origin: https://bitbucket.org/arakiken/w3m/branch/remoteimg (2014-11-16)
+
+	* doc/README.sixel, terms.c: Add README.sixel.  W3M_IMG2SIXEL
+	environmental variable enables to specify options of img2sixel.
+
+	* image.c, terms.c:
+	Add n_terminal_image argument to put_image_{sixel|osc5379}().
+	Use struct winsize to calculate ppc and ppl.
+
+	* terms.c: If SCREEN_VARIANT=sixel on GNU screen, exec img2sixel
+	without -P option.
+
+	* terms.c: ttymode_set() -> ttymode_reset().
+
+	* terms.c: Fix.
+
+	* terms.c: Support GNU screen.
+
+	* terms.c: Show GIF (except animation GIF) correctly.
+
+	* main.c, terms.c: img2sixel exits by Ctrl+C. Enable GIF Animation if
+	'I' is pressed to show it.
+
+	* image.c: Add declaration of get_pixel_per_cell().
+
+	* terms.c: Show the first frame of animation gif files.
+
+	* terms.c: system() -> fork()&execvp()
+
+	* display.c: Draw underline on anchor which contains cboth text and
+	images.
+
+	* etc.c: Remove close_tty() from setup_child() because close_tty()
+	sometimes interrupts loadGeneralFile() in loadImage() and corrupt
+	image data can be cached in ~/.w3m.
+
+	* image.c: Minor fix.
+
+	* image.c: Cache image files if at all possible and convert them to
+	sixel when -sixel option is specified.
+
+	* image.c: Init pixel_per_{char|line}_i if get_pixel_per_cell() fails.
+
+	* display.c, file.c, fm.h, image.c, main.c, terms.c:
+	Add -sixel option which supports image processing by img2sixel.
+
+	* image.c: Don't download image files whose size is specified in
+	<img> tag.
+
+	* image.c: Minor fixes of parseImageHeader().
+
+	* image.c: Determine the format of an image file by its header data
+	not by its file name suffix.
+
+	* image.c: Read width and height from jpeg, png and gif files directly
+	instead of executing w3mimgdisplay -size.
+
+	* display.c: display.c: Draw underline on anchor text which is not
+	overlapped with any image.
+
+	* terms.c: Clear fd_set by FD_ZERO() before select().
+
+	* file.c: nw and ni are rounded up instead of rounded off to show
+	every corner of images.
+
+	* terms.c: Change time to wait for the response of "\x1b[14t\x1b[18t"
+	from 0.1 sec to 0.5 sec.
+
+	* image.c:
+	- clearImage() works.
+	- Use cached image files created by w3m in getImage().
+
+	* file.c: Hack for alignment.
+
+	* fm.h, image.c, terms.c:
+	- Adjust the image size to the terminal cell size.
+	- If the image size is specified in html source, skip to load the image.
+
+	* display.c, fm.h, image.c, main.c, terms.c, w3mimg/x11/x11_w3mimg.c:
+	Support remote image by OSC 5379 show_picture sequence.
+
+2014-12-06  Olaf Hering  <olh@suse.de>
+
+	* parsetagx.c: Fix crash in parse_tag() during every start.
+	Origin: https://build.opensuse.org/package/view_file/openSUSE:Factory/w3m/w3m-parsetagx-crash.patch?expand=1
+
+	* fm.h: Change the default to alt_entity=0.
+	Change the default for the option "Use ASCII equivalents to
+	display entities" from YES to NO.
+	Origin: https://build.opensuse.org/package/view_file/openSUSE:Factory/w3m/w3m-0.5.1-no-ASCII-equivalents-by-default.patch?expand=1
+	Bug-Novell: https://bugzilla.novell.com/show_bug.cgi?id=247397
+
+	* anchor.c, libwc/gb18030.c, libwc/ucs.c, regex.c:
+	Fix a few harmless uninitialized variables.
+	Origin: https://build.opensuse.org/package/view_file/openSUSE:Factory/w3m/w3m-uninitialized.patch?expand=1
+
+2014-12-06  Peter Poeml  <poeml@suse.de>
+
+	* terms.c: Prevent segfault when editing a textarea field with vi.
+	Add fix for segfault that can occur when editing a textarea field
+	with vi, and returning to w3m (it seems to happen if the terminal
+	is not writable, as when using w3m after 'su - some_user')
+	Origin: https://build.opensuse.org/package/view_file/openSUSE:Factory/w3m/w3m-0.4.1-textarea-segfault.dif?expand=1
+
+2014-12-04  Tatsuya Kinoshita  <tats@debian.org>
+
+	* acinclude.m4: Follow updated configure.
+
+2014-12-03  Yusuke Baba  <babayaga1@y8.dion.ne.jp>
+
+	* configure, w3mimg/fb/fb.c, w3mimg/fb/fb.h, w3mimg/fb/fb_w3mimg.c:
+	Support FreeBSD framebuffer.
+	Origin: http://www.ac.auone-net.jp/~baba/w3m-img/index.html
+	Bug-FreeBSD: https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=122673
+
+2014-12-02  Naohiro Aota  <naota@gentoo.org>
+
+	* acinclude.m4, configure, w3mimg/fb/fb_gdkpixbuf.c:
+	* w3mimg/x11/x11_w3mimg.c:
+	Depend on gdk-pixbuf instead of gtk when gtk2.
+	Origin: http://sources.gentoo.org/cgi-bin/viewvc.cgi/gentoo-x86/www-client/w3m/files/w3m-0.5.3-gdk-pixbuf.patch?revision=1.1
+
+2014-12-02  Jeroen Roovers  <jer@gentoo.org>
+
+	* acinclude.m4, configure: Add tinfo to with_termlib.
+	Fix building against sys-libs/ncurses[tinfo].
+	Origin: https://504588.bugs.gentoo.org/attachment.cgi?id=372650
+	Bug-Gentoo: https://bugs.gentoo.org/show_bug.cgi?id=504588
+
+2014-12-01  OBATA Akio  <obache@netbsd.org>
+
+	* acinclude.m4, configure:
+	Assume defined PKG_CONFIG points right location when gtk2.
+	Origin: http://cvsweb.netbsd.org/bsdweb.cgi/pkgsrc/www/w3m/patches/patch-aa?rev=1.13&content-type=text/x-cvsweb-markup
+	Origin: http://cvsweb.netbsd.org/bsdweb.cgi/pkgsrc/www/w3m/patches/patch-ak?rev=1.1&content-type=text/x-cvsweb-markup
+
+2014-12-01  Vsevolod Stakhov  <vsevolod@FreeBSD.org>
+
+	* config.h.in: Disable USE_EGD for LibreSSL.
+	Disable use of RAND_egd as it is absent in FreeBSD.
+	This also fixes build error with LibreSSL.
+	Origin: https://bz-attachments.freebsd.org/attachment.cgi?id=144635
+	Bug-FreeBSD: https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=191852
+	Bug-FreeBSD: https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=191956
+
+2014-12-01  zimous  <zimous@matfyz.cz>
+
+	* po/ja.po: Set Language tag properly for Japanese translation.
+	Origin: https://512722.bugs.gentoo.org/attachment.cgi?id=378452
+	Bug-Gentoo: https://bugs.gentoo.org/show_bug.cgi?id=512722
+
+2014-11-30  Tatsuya Kinoshita  <tats@debian.org>
+
+	* doc/w3m.1: Typo fix.
+
+2014-11-30  Markus Hiereth  <post@hiereth.de>
+
+	* doc/w3m.1: Miscellaneous changes to improve English manpage.
+	Origin: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=766550#30
+
+2014-11-29  Markus Hiereth  <post@hiereth.de>
+
+	* doc/w3m.1: Improve FILES.
+	Origin: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=766550#30
+	Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=403634
+
+	* doc/w3m.1: Improve EXAMPLES.
+	Origin: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=766550#30
+	Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=380560
+
+	* doc/w3m.1: Improve explanation about option -N.
+	Origin: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=766550#30
+	Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=345084
+	Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=530468
+
+	* doc/w3m.1: Note that -cols only affects when HTML is rendered.
+	Origin: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=766550#30
+	Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=285251
+
+	* doc/w3m.1: Add more info on configuration.
+	Origin: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=766550#30
+	Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=268211
+
+2014-11-29  Justin B Rye  <justin.byam.rye@gmail.com>
+
+	* scripts/w3mman/w3mman.1.in: Tweak for W3MMAN_W3M.
+	Origin: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=771003#5
+
+	* scripts/w3mman/w3mman.1.in: English fixes.
+	Origin: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=766550#25
+
+2014-11-29  Tatsuya Kinoshita  <tats@debian.org>
+
+	* version.c.in: Update to 0.5.3+debian-19+.
+
+2014-11-29  Justin B Rye  <justin.byam.rye@gmail.com>
+
+	* scripts/w3mman/w3mman2html.cgi.in: Fix Perl warnings.
+	Origin: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=771004
+
+2014-10-21  Tatsuya Kinoshita  <tats@debian.org>
+
+	* version.c.in: Update to 0.5.3+debian-19
+
+	* po/LINGUAS: Correct LINGUAS to a whitespace separated list
+
+2014-10-21  Markus Hiereth  <markus.hiereth@freenet.de>
+
+	* po/LINGUAS, po/de.po: Add German translation
+	Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=763964
+
+2014-10-15  Tatsuya Kinoshita  <tats@debian.org>
+
+	* version.c.in: Update to 0.5.3+debian-18
+
+	* doc-jp/README.SSL: Update README.SSL to follow default values
+
+	* config.sub: Update config.sub with autotools-dev 20140911.1
+
+	* fm.h: Disable SSLv3 by default [CVE-2014-3566]
+	cf. https://blog.mozilla.org/security/2014/10/14/the-poodle-attack-and-the-end-of-ssl-3-0/
+
+2014-10-15  Ludwig Nussel  <ludwig.nussel@suse.de>
+
+	* fm.h: Force ssl_verify_server on and disable SSLv2 support
+	Origin: http://www.openwall.com/lists/oss-security/2010/06/14/4
+
+2014-10-13  Tatsuya Kinoshita  <tats@debian.org>
+
+	* version.c.in: Update to 0.5.3+debian-17+
+
+2014-10-04  Tatsuya Kinoshita  <tats@debian.org>
+
+	* libwc/ambwidth_map.awk, libwc/map/ucs_ambwidth.map:
+	Fix incorrect generation of ucs_ambwidth_map
+
+2014-08-22  Tatsuya Kinoshita  <tats@debian.org>
+
+	* version.c.in: Update to 0.5.3+debian-17
+
+	* config.guess:
+	Update config.guess to 2014-03-23 with autotools-dev 20140510.1
+
+	* config.sub:
+	Update config.sub to 2014-05-01 with autotools-dev 20140510.1
+
+2014-08-22  Micah Cowan  <micah@addictivecode.org>
+
+	* main.c: Support Boehm GC 7.2
+	Replace Gentoo's patch to prevent segfaults due to infinite recursion.
+	Origin: https://bugs.debian.org/cgi-bin/bugreport.cgi?msg=5;filename=080_gc72.patch;att=1;bug=758831
+	Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=758831
+	Bug-Fedora: https://bugzilla.redhat.com/show_bug.cgi?id=555467
+	Bug: http://sourceforge.net/p/w3m/patches/63/
+	Bug: http://sourceforge.net/p/w3m/patches/59/
+
+2014-08-22  Tatsuya Kinoshita  <tats@debian.org>
+
+	* main.c:
+	Revert "Support Boehm GC 7.2" (w3m-0.5.2-gc72.patch from Gentoo)
+	This reverts commit 4331db3e3e673ac4dbfe8e9f2b42a8e0478dc98a.
+
+2014-06-23  Tatsuya Kinoshita  <tats@debian.org>
+
+	* version.c.in: Update to 0.5.3+debian-16
+
+	* url.c: Disable ciphers that use keys smaller than 128 bits
+	Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/w3m/+bug/1325674
+
+2014-01-04  Tatsuya Kinoshita  <tats@debian.org>
+
+	* version.c.in: Update to 0.5.3+debian-15
+
+2014-01-03  Tatsuya Kinoshita  <tats@debian.org>
+
+	* version.c.in: Update to 0.5.3+debian-14
+
+	* acinclude.m4, configure: Use pkg-config to build with imlib2 1.4.6
+
+	* doc/HISTORY, doc/README.cookie, doc/README.m17n:
+	Prefer US-ASCII rathar than Japanese encodings in English documents
+
+2013-12-27  Tatsuya Kinoshita  <tats@debian.org>
+
+	* doc-jp/MANUAL.html, doc/MANUAL.html:
+	Cleanup unusable links in MANUAL.html
+	Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=517315
+
+	* version.c.in: Update to 0.5.3+debian-13+
+
+2013-12-17  Tatsuya Kinoshita  <tats@debian.org>
+
+	* version.c.in: Update to 0.5.3+debian-13
+
+2013-12-14  Tatsuya Kinoshita  <tats@debian.org>
+
+	* config.guess:
+	Update config.guess to 2013-06-10 with autotools-dev 20130810.1
+
+	* config.sub:
+	Update config.sub to 2013-08-10 with autotools-dev 20130810.1
+
+2013-12-07  Reinhard Max  <max@suse.de>
+
+	* local.c: Fix a directory descriptor leak in loadLocalDir
+	Patch from openSUSE on 2009-09-07.
+	Origin: https://build.opensuse.org/package/view_file/openSUSE:Factory/w3m/w3m-closedir.patch
+	Bug-Novell: https://bugzilla.novell.com/show_bug.cgi?id=531675
+
+2013-12-07  AIDA Shinra  <shinra@j10n.org>
+
+	* main.c: Fix crash after SEARCH_NEXT
+	Patch from <http://www.j10n.org/files/w3m-cvs-1.1055-search-next.patch>,
+	[w3m-dev:04473] on 2013-12-07.
+
+2013-11-11  Paul Boekholt  <p.boekholt@gmail.com>
+
+	* file.c: Add support for single quoted meta refresh URL
+	Bug: https://sourceforge.net/p/w3m/patches/53/
+
+2013-11-07  Cristian Rodriguez  <crrodriguez@opensuse.org>
+
+	* url.c: Use SSL_OP_NO_COMPRESSION if available
+	Due to the "CRIME attack" (CVE-2012-4929) HTTPS clients that
+	negotiate TLS-level compression can be abused for MITM attacks.
+	* url.c: Use SSL_MODE_RELEASE_BUFFERS if available
+	Patch from openSUSE on 2012-11-12:
+	https://build.opensuse.org/request/show/141054
+
+2013-10-15  Tatsuya Kinoshita  <tats@debian.org>
+
+	* Makefile.in:
+	Depend on funcname.tab to fix parallel make issue of scripts
+	Bug: https://sourceforge.net/p/w3m/patches/64/
+	Bug-Gentoo: https://bugs.gentoo.org/show_bug.cgi?id=362249
+
+	* w3mimg/Makefile.in:
+	Avoid prerequisite $(IMGOBJS) to fix parallel make issue of w3mimg
+	Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=726188
+
+	* acinclude.m4, configure:
+	Explicitly add -lX11 to IMGX11LDFLAGS only when gtk2
+	Bug: https://sourceforge.net/p/w3m/patches/57/
+
+	* w3mimg/Makefile.in: Revert "Fix parallel make issue"
+	This reverts commit aa6f871c6dcc108118142bcc786e4a6ac3d46867.
+
+	* Makefile.in:
+	Revert "Explicitly link w3mimgdisplay with -lX11 to build with gcc 4.5"
+	This reverts commit 7410954066d68ac2ad6aea638801714447321fec.
+
+2013-10-14  AIDA Shinra  <shinra@j10n.org>
+
+	* url.c: Define schemeNumToName() to fix scheme bug
+	Patch from <http://www.j10n.org/files/w3m-cvs-1.1055-schemebug.patch>,
+	[w3m-dev:04470] on 2013-10-14.
+	Bug: https://sourceforge.net/p/w3m/patches/60/
+
+	* config.h.in, file.c, fm.h, html.h, image.c, indep.c, indep.h:
+	* istream.c, istream.h, local.c, main.c, mimehead.c, proto.h:
+	Workaround of GC crash on Cygwin64
+	Patch from <http://www.j10n.org/files/w3m-cvs-1.1055-win64gc.patch>,
+	[w3m-dev:04469] on 2013-10-14.
+
+2013-10-14  Tatsuya Kinoshita  <tats@debian.org>
+
+	* version.c.in: Update to 0.5.3+debian-12+
+
+2013-10-14  Jarek Czekalski  <jarekczek@poczta.onet.pl>
+
+	* terms.c: Fix paren in check_cygwin_console()
+	Bug: https://sourceforge.net/p/w3m/patches/66/
+
+2013-10-13  Tatsuya Kinoshita  <tats@debian.org>
+
+	* version.c.in: Update to 0.5.3+debian-12
+
+	* doc-jp/MANUAL.html, doc-jp/w3m.1, doc/MANUAL.html, doc/w3m.1:
+	Update document for the -s option change
+	Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=75527
+
+	* terms.c: Do not fail when LANG is not set
+	Check whether the value of LC_ALL, LC_CTYPE or LANG is not NULL in
+	check_cygwin_console().
+	Bug: https://sourceforge.net/p/w3m/patches/66/
+
+2013-10-12  Tatsuya Kinoshita  <tats@debian.org>
+
+	* table.h: Bump MAXCOL to 256
+	Bug: https://sourceforge.net/p/w3m/feature-requests/24/
+
+2013-10-12  Laurence Richert  <laurencerichert@yahoo.de>
+
+	* main.c, proto.h: vim/-perator like handling
+	- half page scrolling
+	- jumping to elements numbered by getLinkNumberStr() from Karsten
+	  Schoelzel
+	Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=724028
+
+2013-10-12  Tatsuya Kinoshita  <tats@debian.org>
+
+	* doc-jp/README, doc/README:
+	Mention project page rather than unavailable mailing lists
+
+2013-10-09  Rafael Laboissiere  <rafael@laboissiere.net>
+
+	* doc/README.img: Fix typo
+	Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=725892
+
+2013-08-12  Tatsuya Kinoshita  <tats@debian.org>
+
+	* version.c.in: Update to 0.5.3+debian-11+
+
+	* ChangeLog: Update ChangeLog to use contributor's name
+
+2013-08-08  Tatsuya Kinoshita  <tats@debian.org>
+
+	* version.c.in: Update to 0.5.3+debian-11
+
+2013-08-04  Tatsuya Kinoshita  <tats@debian.org>
+
+	* Str.c: Check length for Strchop()
+
+	* main.c: Fix potentially segfault of execdict()
+
+	* version.c.in: Update to 0.5.3+debian-10+
+
+	* file.c: Fix segfault of loadGeneralFile()
+	Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=718612
+
+2013-08-02  Tatsuya Kinoshita  <tats@debian.org>
+
+	* version.c.in: Update to 0.5.3+debian-10
+
+2013-08-02  Piotr P. Karwasz <piotr.p@karwasz.org>
+
+	* scripts/w3mman/w3mman2html.cgi.in:
+	Correct underline processing and more UTF-8 support for w3mman2html.cgi
+	Patch from <https://bugs.launchpad.net/ubuntu/+source/w3m/+bug/680202>
+	on 2010-11-23.
+
+2013-08-01  Hilko Bengen  <bengen@debian.org>
+
+	* entity.c: Ignore SOFT HYPHEN to prevent drawing hyphens everywhere
+	Patch from <http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=441934>
+	on 2011-03-01.
+
+2013-08-01  Tatsuya Kinoshita  <tats@debian.org>
+
+	* doc-jp/README, doc/README: Update contact list in README
+	Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=696209
+
+2013-07-30  Tatsuya Kinoshita  <tats@debian.org>
+
+	* config.guess, config.sub:
+	Update config.guess and config.sub to supprot aarch64
+	Updated with Debian autotools-dev version 20130515.1.
+
+2013-07-30  Conrad J.C. Hughes  <debbugs@xrad.org>
+
+	* main.c: Sort anchors by sequence number in -dump
+	Patch from <http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=657666>
+	on 2012-01-27.
+
+2013-07-30  Tatsuya Kinoshita  <tats@debian.org>
+
+	* version.c.in: Update to 0.5.3+debian-9+
+
+2013-07-29  Tatsuya Kinoshita  <tats@debian.org>
+
+	* version.c.in: Update version to w3m/0.5.3+debian-9
+
+	* version.c.in: Set CURRENT_VERSION to debian version
+
+2013-07-28  Tatsuya Kinoshita  <tats@debian.org>
+
+	* file.c: Fix segfault of process_button()
+
+2013-04-08  AIDA Shinra  <shinra@j10n.org>
+
+	* file.c: One more patch for siteconf from [w3m-dev 04464]
+
+	* anchor.c, config.h.in, display.c, doc-jp/README.siteconf:
+	* doc/README.siteconf, file.c, fm.h, form.c, frame.c, func.c:
+	* history.c, indep.c, indep.h, linein.c, main.c, map.c, menu.c:
+	* po/ja.po, proto.h, rc.c, url.c: Support the siteconf feature
+	Patch to support the siteconf feature, from [w3m-dev 04463]
+	on 2012-06-27.
+
+2013-04-08  Hayaki Saito  <user@zuse.jp>
+
+	* keybind.c, main.c, proto.h, terms.c:
+	Support SGR 1006 mouse reporting
+	Patch to support SGR 1006 mouse reporting, from [w3m-dev 04466]
+	on 2012-07-15.
+
+2012-05-19  Hilko Bengen  <bengen@debian.org>
+
+	* form.c: Assume "text" if an input type is unknown
+	Patch from <http://bugs.debian.org/615843> on 2011-03-01.
+
+2012-05-19  Simon Ruderich  <simon@ruderich.org>
+
+	* Makefile.in: Use $(CPPFLAGS) with $(CPP)
+	Patch from <http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=665491>
+	on 2012-03-24.
+
+2012-05-03  Miroslav Šulc  <fordfrog@gentoo.org>
+
+	* w3mimg/Makefile.in: Fix parallel make issue
+	Patch from Gentoo
+	<http://sources.gentoo.org/cgi-bin/viewvc.cgi/gentoo-x86/www-client/w3m/files/w3m-0.5.3-parallel-make.patch?revision=1.1&view=markup>
+	<https://bugs.gentoo.org/show_bug.cgi?id=353390> on 2011-02-01.
+
+2012-05-03  MATSUU Takuto  <matsuu@gentoo.org>
+
+	* main.c: Support Boehm GC 7.2
+	Patch from Gentoo
+	<http://sources.gentoo.org/cgi-bin/viewvc.cgi/gentoo-x86/www-client/w3m/files/w3m-0.5.2-gc72.patch?revision=1.1&view=markup>
+	on 2009-12-13.
+
+2012-05-02  Reinhard Tartler  <siretart@tauware.de>
+
+	* istream.c, istream.h:
+	Fix that struct file_handle conflicts with glibc 2.14
+	Patch from <https://bugs.launchpad.net/ubuntu/+source/w3m/+bug/935540>
+	on 2012-02-19.
+
+2011-10-30  Colin Watson  <cjwatson@ubuntu.com>
+
+	* acinclude.m4, configure, w3mbookmark.c:
+	Appease gcc -Werror=format-security.
+	Patch from 0.5.3-3ubuntu1 on 2011-10-23.
+	Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=646321
+
+2011-06-19  Martin Pitt  <martin.pitt@ubuntu.com>
+
+	* Makefile.in:
+	Explicitly link w3mimgdisplay with -lX11 to build with gcc 4.5
+	Patch from 0.5.2-10ubuntu1 on 2010-12-03.
+	Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=605761
+
+2011-06-19  Fumitoshi UKAI  <ukai@debian.or.jp>
+
+	* main.c:
+	Change the -s option to "squeeze multiple blank lines" for pager
+	Change the -s option from "display charset Shift_JIS" to "squeeze
+	multiple blank lines" to work as /usr/bin/pager.  In addition, the
+	options -j and -e are disabled.  To specify the display charset,
+	use -O{s|j|e} instead.
+	Patch from [w3m-dev 01275] on 2000-10-26.
+	Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=75527
+
+2011-06-19  Hiroyuki Ito  <ZXB01226@nifty.com>
+
+	* file.c, fm.h, html.c, html.h, proto.h, table.c, tagtable.tab:
+	Support the button element as defined in HTML 4.01
+	Patch from upstream, [w3m-dev 04411] on 2010-09-17, to support the
+	button element.  It is discussed upstream and incomplete, but enough
+	to login Launchpad.
+	Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=136810
+
 2012-05-22  Tatsuya Kinoshita <tats@vega.ocn.ne.jp>
 
 	* [w3m-dev 04451] w3m/entity.h should be removed when `make clean'
